core/src/main/java/dev/streamx/cli/license/LicenseProcessorEntrypoint.java

```diff
@@ -65,13 +71,28 @@ private void proceedLicenseAcceptance(LicenseSettings licenseSettings, LocalDate
     print("");
     print("Do you accept the license agreement? [Y/n]");
 
+    if (licenseContext.isAcceptLicenseFlagSet()) {
+      proceedAutomaticLicenseAcceptance(licenseSettings, now);
+      return;
+    }
+
     if (acceptingStrategy.isLicenseAccepted()) {
       licenseSettingsStore.acceptLicense(licenseSettings, now);
     } else {
       throw LicenseException.licenseAcceptanceRejectedException();
     }
   }
 
+  private void proceedAutomaticLicenseAcceptance(LicenseSettings licenseSettings,
+      LocalDateTime now) {
+    print("Y -> \"--accept-license\" was passed");
+    if (licenseSettings.lastLicenseFetch().isPresent()) {
```

Comment:
Nope. In case of passing flag "--accept-license" and fetch failed/timeouted, `lastLicenseFetch()` can be empty.
So data about currently accepted license is stored if fetch is successful. 
But if fetch is timeouted, it skips storing information about accepted license as we do not know what we accepted.

`licenseSettingsStore.acceptLicense` requires `lastLicenseFetch()` data.